### PR TITLE
Update installation guide

### PR DESCRIPTION
**Pull request by eliaspekkala**
This pull request tries to make the Getting Started guide more helpful. When I followed the guide on Linux, I hit ~5 undocumented errors (missing packages/dependencies), so I had to spend time troubleshooting and Googling to get Odin to compile. I was also a bit confused about the structure of the document, so I tried to improve that. My hope is that if the Getting Started guide is easier to follow, more people try Odin and won't give up if they run into compilation errors.

Changes:
- Restructure the document.
- Give more info about PATH.
- Add some more notes on Linux (helps with troubleshooting).
- Separate Linux and BSD guide.
- Combine Requirements section with Platform section.

### CODE REVIEW

**Comment by eliaspekkala** [comment]
Can we make it so that the platform-specific steps can be folded or expanded/hidden? Is there Markdown syntax for that?

File: content/docs/install.md [hunk 1]
```diff
@@ -1,86 +1,98 @@
...
+- Install [Homebrew](https://brew.sh/).
+- Install LLVM through Homebrew: `brew install llvm@14`.
+- [Add the LLVM binaries to your `$PATH`.](#add-llvm-to-your-path)
+- Navigate to the Odin directory inside your terminal.
+- Run `make` to build Odin. This will create an Odin binary.
+- [Add the Odin directory to your `$PATH`.](#add-odin-to-your-path)
+- Use `odin --help` to learn how to run your first Odin program.
 
-**Note**: Cloning the repository is recommended in order to make [updating](#updating-the-compiler) easier.
+**Note:** On newer versions of MacOS, some headers are not installed by default. Open `macOS_SDK_headers_for_macOS_*.pkg` in `/Library/Developer/CommandLineTools/Packages/` to install the ones you need.
```
Comment (eliaspekkala) [inline content/docs/install.md:50]: I don't really understand this. Was it a correct edit?

**Comment by eliaspekkala** [comment]
Sorry, the diff became really scrambled. Maybe just view the new version when reviewing?

**Comment by eliaspekkala** [comment]
Also, I removed the Support section. Thought it would be obvious which operating systems are supported by the instructions. Though, are the notes about AMD64 and ARM64 support needed?

File: content/docs/install.md [hunk 2]
```diff
@@ -1,86 +1,94 @@
...
 
+- Choose either a release build or a nightly build.
+- Download the build that is specific to your platform.
+- Unzip it and move the directory to your desired location.
+- [Add the Odin directory to your `$PATH`.](#add-odin-to-your-path)
+- Use `odin --help` to learn how to run your first Odin program.
 
-## Clone or download Odin binaries
-You can either:
+## Clone and build Odin
```
Comment (gingerBill) [inline content/docs/install.md:22]: It needs to be made clear that you can either Download Odin **or** Clone and build Odin, as how it currently laid out, it looks you need to do both.

File: content/docs/install.md [hunk 3]
```diff
@@ -1,86 +1,94 @@
...
+### BSD
+- On FreeBSD, run: `pkg install bash git llvm14`
+- [Add the LLVM binaries to your `$PATH`.](#add-llvm-to-your-path)
+- Navigate to the Odin directory inside your terminal.
+- Run `make` to build Odin. That creates the `odin` binary.
+- [Add the Odin directory to your `$PATH`.](#add-odin-to-your-path)
+- Use `odin --help` to learn how to run your first Odin program.
 
-To use Odin `link.exe` is required to be in the PATH of the callee as mentioned, this can either be achieved but calling Odin from the X64 Visual Studio command prompt or by calling the vcvarsall.bat (with x64 as an argument) script either in your shell or in your build script.
+## Add LLVM to your PATH
```
Comment (gingerBill) [inline content/docs/install.md:80]: This is only required on non-Windows systems.

File: content/docs/install.md [hunk 4]
```diff
@@ -1,86 +1,94 @@
...
+To add LLVM to the `$PATH` environment variable, insert the line: `export PATH=$PATH:/path/to/llvm/bin` in your shell configuration file. This file may be called `~/.bashrc` or `~/.zshrc` or `/etc/profile`; it depends on your shell and platform. Then apply the change to the current session by running: `source ~/.name_of_shell_config_file`. Alternatively, you can restart the terminal to apply the change.
 
-### For MacOS
-Make sure all requirements for MacOS are installed, after installing LLVM through Homebrew make sure to add it to the PATH:
-- run `echo 'export PATH="/usr/local/opt/llvm/bin:$PATH"' >> ~/.zshrc_profile` to add LLVM to your PATH.
+**Note:** The installed location of LLVM depends on your platform. Two possible locations to look into are: `/usr/local/opt/llvm/bin` and `/usr/lib/llvm14/bin`.
 
-Then run `source ~/.bash_profile` or `source ~/.zshrc` to update your PATH variable in the current terminal session depending on your shell.
+## Add Odin to your PATH
```
Comment (gingerBill) [inline content/docs/install.md:85]: This is not necessary, but can be useful. Make that clear.

File: content/docs/install.md (hunk 2 above)
Comment (eliaspekkala) [reply content/docs/install.md]: I updated the wording, section names, and structure to make that clear.

File: content/docs/install.md (hunk 3 above)
Comment (eliaspekkala) [reply content/docs/install.md]: I removed that section and instead made it a step for all platforms that require it.

File: content/docs/install.md (hunk 4 above)
Comment (eliaspekkala) [reply content/docs/install.md]: I made this information a small tip instead.

**Pull request closed without merging**